### PR TITLE
Make sure the summary loaded.

### PR DESCRIPTION
**Pull request by joakim-hove**
**Task**
In the case we do not have any summary variables present there has been a very strong tendency to *not* load the summary data. These are potentially needed for the time mapping, and we should load them if possible.

Final attempt to fix WPRO issue in 2.2/ the first commit should be cherry-picked to 2.2

It works on my machine ..... 

#123 was the first attempt to fix this issue. Now it even comes with a test.

**Pre un-WIP checklist**
- [x] Statoil tests pass locally
- [ ] Have completed graphical integration test steps

### CODE REVIEW

**Comment by pgdr** [comment]
Did the Statoil tests pass locally, and the UI test pass?

File: libenkf/src/enkf_state.c [hunk 1]
```diff
@@ -794,7 +800,9 @@ int enkf_state_load_from_forward_model(enkf_state_type * enkf_state ,
   if (ensemble_config_have_forward_init( enkf_state->ensemble_config ))
     result |= enkf_state_forward_init( enkf_state , run_arg );
 
+  printf("AFter forward:%d \n",result);
   result |= enkf_state_internalize_results( enkf_state , run_arg , msg_list );
+  printf("AFter internalize:%d \n",result);
```
Comment (pgdr) [inline libenkf/src/enkf_state.c]: Why are the Fs capitalized?

File: libenkf/src/enkf_state.c [hunk 2]
```diff
@@ -706,15 +706,21 @@ static void enkf_state_internalize_GEN_DATA(enkf_state_type * enkf_state ,
...
-  if (!load_summary) {
+  bool load_summary = false;
+
+  if (ensemble_config_has_GEN_DATA(state->ensemble_config))
+    load_summary = true;
+
+  if (ensemble_config_has_impl_type(state->ensemble_config, SUMMARY))
+    load_summary = true;
+
+  {
```
Comment (pgdr) [inline libenkf/src/enkf_state.c]: Please remove these guards surrounding and immediately succeeding `if`s.  They are very confusing (and even more unnecessary).

File: libenkf/tests/enkf_workflow_job_test2.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright (C) 2013  Statoil ASA, Norway.
+
+   The file 'enkf_workflow_job_test.c' is part of ERT - Ensemble based Reservoir Tool.
```
Comment (pgdr) [inline libenkf/tests/enkf_workflow_job_test2.c]: The header is wrong, please update.

File: libenkf/tests/enkf_workflow_job_test2.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  stringlist_append_copy( args , "1");
+  test_assert_true( ert_test_context_run_worklow_job( test_context , job_name , args) );
+  stringlist_free( args );
+
+  {
+    enkf_main_type * enkf_main = ert_test_context_get_main( test_context );
+    enkf_fs_type * fs = enkf_main_get_fs( enkf_main );
+    time_map_type * time_map = enkf_fs_get_time_map( fs );
+    test_assert_true( time_map_get_last_step( time_map ) > 0 );
+  }
```
Comment (pgdr) [inline libenkf/tests/enkf_workflow_job_test2.c]: why is this inside `{}`?

**Comment by joakim-hove** [comment]
> Did the Statoil tests pass locally, and the UI test pass?

I have been working with this more-or-less all morning and do not understand:'

1. Everything - including Statoil tests and the gui test works *on my machine*.
2. There is a test-failure on Travis and the same test fails on the internal Jenkins build server.

It can seem like it is some file syncronization issue, but I do not understand this just yet.

File: libenkf/src/enkf_state.c (hunk 1 above)
Comment (joakim-hove) [reply libenkf/src/enkf_state.c]: This is debug output - it should go entirely.

File: libenkf/tests/enkf_workflow_job_test2.c (hunk 3 above)
Comment (joakim-hove) [reply libenkf/tests/enkf_workflow_job_test2.c]: Done

File: libenkf/src/enkf_state.c (hunk 2 above)
Comment (joakim-hove) [reply libenkf/src/enkf_state.c]: Well - you think they are confusing; I think they improve readability; these are subjective matters.

**Comment by pgdr** [comment]
It seemed one of the Travis jobs were hanging ... I restarted it, let's see what the outcome is.

**Comment by pgdr** [comment]
The same test fails at my machine on your branch.

**Comment by pgdr** [comment]
There are some logs and binary files now (8000+ lines?), including jobs.json containing `/private/joaho/work/ERT/libres/test-data/...`.   Can you add only the necessary files?

**Review by pgdr — APPROVED** [review]
Merge at will after cleaning test-data

**Comment by joakim-hove** [comment]
> Can you add only the necessary files?

yes - I have just done that.

**Pull request merged**